data/campaigns/World_Conquest/scenarios/WC_II_scenario.cfg

```diff
@@ -22,22 +22,89 @@ _ "World Conquest 3p" #enddef
     _ "A randomly generated cooperative campaign for 3 players. It has 6 levels of difficulty.
 (Expert level, 5 scenarios.)" #enddef
 
+#define WC2_HUMAN_DIFFICULTY IMAGE COLOR
+"units/{IMAGE}.png~RC(magenta>{COLOR})"#enddef
+
+#define WC2_NIGHTMARE_DIFFICULTY
+"units/monsters/fire-dragon.png~CROP(0,0,160,160)~SCALE_INTO(72,72)~RC(magenta>red)"#enddef
+
+#define WC2_CAMPAIGN_DIFFICULTY DIFFICULTY IMAGE LABEL DESCRIPTION POWER TRAINED HEROES GOLD TRAIN XP
+    {CAMPAIGN_DIFFICULTY {DIFFICULTY} {IMAGE} {LABEL} {DESCRIPTION}}
+    [event]
+        name=preload
+        # These events run for every difficulty, so this Lua block ensures only the active difficulty's
+        # event actually fills the settings variable.
+        [lua]
+            code=<<
+                if wesnoth.scenario.difficulty ~= (...).difficulty then
+                    wesnoth.wml_actions["return"]{}
+                end
+            >>
+            [args]
+                difficulty={DIFFICULTY}
+            [/args]
+        [/lua]
+        [set_variables]
+            name=wc2_difficulty
+            [value]
+                name={LABEL}
+                enemy_power={POWER}
+                enemy_trained={TRAINED}
+                heroes={HEROES}
+                # adjust bonus gold for number of players
+                extra_gold=$(gold * 2 ^ (3 - nplayers) where nplayers = $wc2_player_count, gold = {GOLD})
+                extra_training={TRAIN}
+                experience_penalty={XP}
+            [/value]
+        [/set_variables]
+    [/event]
+#enddef
+
 #define WC2_CAMPAIGN_NEW PLAYERS
-    ## Everything that usually goes into [scenario] like [event], [load_ressource], map_data
-    ## etc is generated by lua. This is also why there is only one [multiplayer] tag: the next
-    ## scenario is again "WC_II_{PLAYERS}p" and the lua code detects that is was called a second
-    ## time and generated the second scenario. (same for third, fourth and fifth). All the lua
-    ## code inside ./lua/map does map/scenario generation.
-    [multiplayer]
+    [campaign]
         id = "WC_II_{PLAYERS}p"
+        define = CAMPAIGN_WC_{PLAYERS}P
         name= {WC_II_CAMPAIGN_NAME_{PLAYERS}P}
         description = {WC_II_CAMPAIGN_DESC_{PLAYERS}P}
-        allow_new_game=yes
-        require_scenario=no
+        first_scenario = "WC_II_{PLAYERS}p"
+        min_players={PLAYERS}
+        max_players={PLAYERS}
+        type = mp
+        abbrev = _ "WC"
```

Comment:
As far as I can tell, this is the sole _new_ string in here. I think leaving it untranslated for 1.16.0 is probably an acceptable compromise? It could become translated as of 1.16.1.

Side note: Should it include the player count?

Reply:
> Side note: Should it include the player count?

I think the abbrev should be dependent on the number of players, so `WC1p`, 'WC2p' and 'WC3p' would be nice. WC for all might become confusing when viewing save files.